cpp/dolfin/la/SLEPcEigenSolver.h

```diff
@@ -126,11 +126,11 @@ class SLEPcEigenSolver : public common::Variable
   void solve(std::int64_t n);
 
   /// Get ith eigenvalue
-  void get_eigenvalue(double& lr, double& lc, std::size_t i) const;
+  void get_eigenvalue(PetscScalar& lr, PetscScalar& lc, std::size_t i) const;
 
   /// Get ith eigenpair
-  void get_eigenpair(double& lr, double& lc, PETScVector& r, PETScVector& c,
-                     std::size_t i) const;
+  void get_eigenpair(PetscScalar& lr, PetscScalar& lc, PETScVector& r,
+                     PETScVector& c, std::size_t i) const;
```

Comment:
Could we change this to:
```
std::tuple<std::complex, PETScVector, PETScVector> get_eigenpair(std::size_t i) const;
```
Does SLEPc create the Vec for the eigenvector (which would avoid copies)? Maybe not?

This change needs some thought.

Reply:
Forget this suggestion. Doesn't map well onto SLEPc.

Reply:
It makes sense to change only `get_eigenvalue`?

In PETSc-complex the eigenvalue is stored directly in lr and the eigenvector in Vr.  The imaginary part is set to zero.
Since PETSc doesn't support pointers to real or imaginary part of a complex vector, maybe we could overlad these functions:
```
void get_eigenpair(double& lr, double& lc, PETScVector& r, PETScVector& c, std::size_t i) const;
void get_eigenpair(std::complex<double>& l, PETScVector& V, std::size_t i) const;

```
```
void get_eigenvalue(std::complex<double> l, std::size_t i) const;
void get_eigenvalue(double& lr, double& lc, std::size_t i) const;
```

Reply:
Ok..Updated also python wrapper.